ospfd/ospf_vty.c

```diff
@@ -11137,25 +11044,35 @@ static int show_ip_ospf_route_common(struct vty *vty, struct ospf *ospf,
 
 	if (ospf->new_table == NULL) {
 		vty_out(vty, "No OSPF routing information exist\n");
+
+		if (use_vrf)
+			json_object_free(json_vrf);
+
 		return CMD_SUCCESS;
 	}
 
+	if (json)
+		json_routes = json_object_new_object();
+
 	/* Show Network routes. */
-	show_ip_ospf_route_network(vty, ospf, ospf->new_table, json_vrf);
+	show_ip_ospf_route_network(vty, ospf, ospf->new_table, json_routes);
 
 	/* Show Router routes. */
-	show_ip_ospf_route_router(vty, ospf, ospf->new_rtrs, json_vrf);
+	show_ip_ospf_route_router(vty, ospf, ospf->new_rtrs, json_routes);
 
 	/* Show AS External routes. */
 	show_ip_ospf_route_external(vty, ospf, ospf->old_external_route,
-				    json_vrf);
+				    json_routes);
 
 	if (json) {
+		json_object_object_add(json_vrf, "routes", json_routes);
```

Comment:
Code looks great now just one final question: why add the `route` dict?

It is changing the JSON command output and it will break parsers (like the test you adapted).

Reply:
hi @Rafael, 
 "routes" parent key is needed in our  product to get the json o/p with specific filters . To validate  a specific route parameter information from json o/p.
Anyhow, it will look organised if we have all these routes under a parent json key. So added this key.

Reply:
@rzalamena One more reason to add this key is to make this output inline with show ip route. That is why we decided to add this one. With respect to current scripts in ci we have modified all those applicable.

Reply:
We might add this as a duplicate under `routes` and add a deprecation cycle for the output not under `routes`?